src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,264 @@
+
+Topics do not need to have a one-to-one relationship. For example, it is possible for a topic to have multiple Subscribers, which allows a single node to communicate with other nodes.
+
+Also, Publishers and Subscribers do not need to be in separate files. In fact, if the Publisher and Subscriber rely on each other's data, it may be better to keep them in the same file. However, a node should not publish to itself as this would be incredibly inefficient. A way to do this properly would be to have the Subscriber receive data from one topic, the data to be manipulated in some way, and the Publisher to send the result on another topic. 
+
+An example of a Publisher and Subscriber residing in the same file will be provided in the next section.
+
+## Setting up a Publisher and Subscriber
+Within the directory /workspaces/WRoverPlayground/src/topic_training/src, you will find two files called Demo(.cpp or .py). These files will show a basic class containing both a Publisher and Subscriber. There will be comments throughout the files that explain the purpose of specific sections of the code.
+
+Note that there are a few sections of code that are marked as "Optional." You do not need to include everything in the examples when creating your own code. They are simply showing one way of setting up a Publisher and Subscriber in the same file.
+
+One file will be written in C++, and the other will be in Python. It is recommended that you choose the language you are most comfortable with to complete the training. Afterwards, you should discuss with one of the Software Leads to determine which part of the rover you would like to be assigned to. You will be given small tasks first, working your way up to more difficult projects. If these tasks/projects require a different language, then you can look through this training again or take what you know from the first time and learn as you go.
+
+Conintue reading here and go into one of the Demo files based on the language you chose. Make sure you understand what is being said. The general idea of the Demo (.cpp or .py) file is that it takes in data from a separate Publisher file (PlotDemo.cpp or .py) as a float on the topic `plot` and sends that data in the form of a String on the topic `chatter`.
+
+### C++
+
+A basic setup for ROS topics in C++ would look something like this:
+
+``` C++
+ros::init(argc, argv, "Demo");
+ros::NodeHandle nhandle;
+ros::Subscriber sub = nhandle.subscribe("plot", QUEUE_SIZE, chatterCallback);
+ros::Publisher pub = nhandle.advertise<std_msgs::String>("chatter", QUEUE_SIZE);
+```
+
+To break this down,
+
+``` C++
+ros::init(argc, argv, "Demo");
+```
+Initialize ROS and specify the name of the node (which is the name of the file in this case). This requires the parameters from the main function header (argc and argv).
+
+``` C++
+ros::NodeHandle nhandle;
+```
+Create a handle for the node. This will do the actually initialization of the node in the next few lines.
```

Comment:
Please correct me if I misremembered, but the node is initialized using `ros::init(...)`. Here the `NodeHandle` allows the programmer to create subscriber and published objects, which are more like "parts" or "functions" of a node.

Reply:
You're right. I worded it weirdly. I meant that new members will be using it to initialize the Subscriber and Publisher objects in the next couple of lines.

Reply:
Fixed in 30181b7.